learning/step_by_step/godot_design_philosophy.rst

```diff
@@ -0,0 +1,145 @@
+except you’re free to design your scene using the editor, using only the
+code, or mixing and matching the two.
+
+It’s different from prefabs you find in several 3d engines as you can
+inherit from and extend scenes. For example you may create a Magician
+that extends your Character scene. Edit the Character and the Magician
+will update as well. It helps you build your projects so that their
+structure match the game’s design.
+
+|image0|
+
+Also note that Godot offers many different types of objects called
+nodes, each with a specific purpose. Nodes are part of a tree and always
+inherit from their parents up to the Node class. Although the engine
+does feature some components like collision shapes, they’re the
+exception, not the norm.
+
+|image1|
+
+Sprite is a Node2D, a CanvasItem and a Node. It has all the properties
+and features of its three parent classes, like transforms or the ability
+to draw custom shapes and render with a custom shader.
+
+All-inclusive package
+---------------------
+
+Godot tries to provide its own tools to answer all of the most common
+needs. It has a dedicated scripting workspace, an animation editor, a
+tilemap editor, a shader editor, etc.
+
+|image2|
+
+The goal is to offer a full package to create games and a continuous
+user experience. You can still work with external programs as long as
+there is an import plugin for it or you can create one, like the `Tiled
+map editor importer <https://github.com/vnen/godot-tiled-importer>`__.
+
+That is also partly why Godot offers its own programming languages
+GDscript and VisualScript, along with C#. They’re designed for the needs
+of game developers, game designers, and they’re tightly integrated in
+the engine and the editor.
+
+|image3|
+
+*VisualScript is a node-based programming language that integrates well
+in the editor. You can drag and drop nodes or resources into the graph
+to create new code blocks.*
+
+Note that if this is true for 2d at the time of writing, the 3d
+workspace still doesn’t feature as many tools. You’ll need external
+programs or add-ons to edit terrains, animate complex characters, etc.
+Godot provides a complete API to extend the editor’s functionality using
+game code. See `The Godot editor is a Godot game <#>`__ below.
+
+|image4|
+
+*A State Machine editor plugin in Godot 2 by kubecz3k. It lets you
+manage states and transitions visually*
+
+Open source
+-----------
+
+Godot offers a fully open source code-base under the MIT license. This
+means all the technologies that ship with it have to be Free as well.
+For the most part, they’re coded from the ground up by contributors.
+
+Anyone can plug in proprietary tools for the needs of their projects.
+They just won’t ship with the engine. This may include NViDia PhysX,
+Google Admob or an FBX file importer. Any of these can come as
+third-party plugins instead.
+
+On the other hand, an open codebase means you can learn from and extend
+the engine to your heart’s content.
```

Comment:
It *also* means you can debug your game way more easily because all the source (and debug symbols) are available, you can always look under the hood and figure things out and fix them if needed